src/services/products.service.ts

```diff
@@ -3,66 +3,277 @@ import { UserType } from '../core/users/User';
+    const antivirusFeatures = tier.featuresPerService[Service.Antivirus];
+    if (antivirusFeatures.enabled && !mergedFeatures.antivirus.enabled) {
+      mergedFeatures.antivirus = {
+        enabled: true,
+        sourceTierId: tier.id,
+      };
+    }
+  }
+
+  private mergeBackupsFeatures(tier: Tier, mergedFeatures: MergedTierFeatures): void {
+    const backupsFeatures = tier.featuresPerService[Service.Backups];
+    if (backupsFeatures.enabled && !mergedFeatures.backups.enabled) {
+      mergedFeatures.backups = {
+        enabled: true,
+        sourceTierId: tier.id,
+      };
+    }
+  }
+
+  private mergeFeatures(tiers: Tier[]): MergedTierFeatures {
+    const mergedFeatures: MergedTierFeatures = {
+      mail: { enabled: false, addressesPerUser: 0 },
+      meet: { enabled: false, paxPerCall: 0 },
+      vpn: { enabled: false, featureId: '' },
+      antivirus: { enabled: false },
+      backups: { enabled: false },
+      drive: { tier: tiers[0] },
+    };
+
+    const businessTiers = tiers.filter((tier) => tier.featuresPerService[Service.Drive].workspaces.enabled);
+    const individualTiers = tiers.filter((tier) => !tier.featuresPerService[Service.Drive].workspaces.enabled);
+
+    mergedFeatures.drive.tier = this.selectBestDriveTier(businessTiers, individualTiers);
+
+    this.mergeNonDriveFeatures(tiers, mergedFeatures);
+
+    return mergedFeatures;
+  }
+
+  private async collectAllAvailableTiers(userUuid: string, ownersId?: string[]): Promise<Tier[]> {
+    const availableTiers: Tier[] = [];
+
+    const user = await this.usersService.findUserByUuid(userUuid);
+    const userDirectTiers = await this.tiersService.getTiersProductsByUserId(user.id);
+    availableTiers.push(...userDirectTiers);
+
+    if (ownersId && ownersId.length > 0) {
+      const ownerTierPromises = ownersId.map(async (ownerUuid) => {
+        try {
+          const owner = await this.usersService.findUserByUuid(ownerUuid);
+          const ownerTiers = await this.tiersService.getTiersProductsByUserId(owner.id);
+          return ownerTiers.filter((tier) => tier.featuresPerService[Service.Drive].workspaces.enabled);
+        } catch (error) {
+          if (error instanceof UserNotFoundError) return [];
+          throw error;
+        }
+      });
+
+      const ownerTiersArrays = await Promise.all(ownerTierPromises);
+      const businessTiers = ownerTiersArrays.flat();
+      console.log('businessTiers', businessTiers);
+      availableTiers.push(...businessTiers);
+    }
+
+    const uniqueTiers = availableTiers.filter((tier, index, self) => index === self.findIndex((t) => t.id === tier.id));
+
+    return uniqueTiers;
+  }
+
+  async getApplicableTierForUser({ userUuid, ownersId }: { userUuid: string; ownersId?: string[] }): Promise<Tier> {
+    const availableTiers = await this.collectAllAvailableTiers(userUuid, ownersId);
+
+    if (availableTiers.length === 0) {
+      throw new TierNotFoundError(`No tiers found for user uuid ${userUuid}`);
```

Comment:
Just use `NotFoundError`. It already triggers a 404 thanks to the global middleware we added (works like in NestJS).

References:
- [NotFoundError](https://github.com/internxt/payments-server/blob/master/src/errors/Errors.ts#L9).
- [Error handler](https://github.com/internxt/payments-server/blob/master/src/plugins/error-handler.ts)